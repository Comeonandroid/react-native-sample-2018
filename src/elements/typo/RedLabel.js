import React from 'react'
import styled from 'styled-components'
import { prop } from 'styled-tools'
import { PropTypes } from 'prop-types'

import { Text } from '../index'


const RedLabel = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

RedLabel.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  color: ${ prop('theme.colors.danger')};
  font-family: ${ prop('theme.typo.condensedFont') };
  font-size: 12px;
  text-align: center;
`

export default RedLabel