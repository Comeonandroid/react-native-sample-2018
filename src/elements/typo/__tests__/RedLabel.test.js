import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import RedLabel from '../RedLabel'

describe('<RedLabel />', () => {
  itBehavesLike('aThemedComponent', { subject: <RedLabel /> })
  itBehavesLike('aComponentWithChildren', { subject: <RedLabel /> })
})
