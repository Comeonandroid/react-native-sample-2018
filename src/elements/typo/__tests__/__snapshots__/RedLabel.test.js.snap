// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RedLabel /> dark: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fcfcfc",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#f25c74",
          "fontFamily": "Roboto Condensed",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<RedLabel /> light: renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "Open Sans",
      },
      Array [
        Object {
          "color": "#f25c74",
          "fontFamily": "Roboto Condensed",
          "fontSize": 12,
          "textAlign": "center",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<RedLabel /> renders component when passed in 1`] = `
<Styled(Text)>
  <Component />
</Styled(Text)>
`;

exports[`<RedLabel /> renders text when passed in 1`] = `
<Styled(Text)>
  Test
</Styled(Text)>
`;
